packages/mdc-textfield/outline/README.md

```diff
@@ -21,22 +21,64 @@ The outline is a border around all sides of the text field. This is used for the
 
 ## Usage
 
-#### MDCTextFieldOutline API
+### HTML Structure
 
-##### MDCTextFieldOutline.foundation
+```html
+<div class="mdc-text-field__outline">
+  <svg>
+    <path class="mdc-text-field__outline-path"/>
+  </svg>
+</div>
+<div class="mdc-text-field__idle-outline"></div>
+```
 
-MDCTextFieldOutlineFoundation. This allows the parent MDCTextField component to access the public methods on the MDCTextFieldOutlineFoundation class.
+### Usage within `mdc-text-field`
 
-### Using the foundation class
+```html
+<div class="mdc-text-field mdc-text-field--outlined">
+  <input class="mdc-text-field__input" id="my-text-field-id" type="text">
+  <label class="mdc-text-field__label" for="my-text-field-id">Label</label>
+  <div class="mdc-text-field__outline">
+    <svg>
+      <path class="mdc-text-field__outline-path"/>
+    </svg>
+  </div>
+  <div class="mdc-text-field__idle-outline"></div>
+</div>
+```
+
+### CSS Classes
+
+CSS Class | Description
+--- | ---
+`mdc-text-field__outline` | Mandatory
+`mdc-text-field__outline-path` | Mandatory
+`mdc-text-field__idle-outline` | Mandatory
+
+#### `MDCTextFieldOutline`
 
 Method Signature | Description
 --- | ---
-getWidth() => number | Returns the width of the outline element
-getHeight() => number | Returns the height of the outline element
-setOutlinePathAttr(value: string) => void | Sets the "d" attribute of the outline element's SVG path
+`MDCTextFieldOutline.createRipple(rippleFactory, rippleInputMethods) => MDCRipple` | Returns an `MDCRipple` instance set on the `mdc-text-field__outline` element
+
+##### `MDCTextFieldOutline.foundation`
+
+This allows the parent `MDCTextField` component to access the public methods on the `MDCTextFieldOutlineFoundation` class.
+
+##### `MDCTextFieldOutline.createRipple(rippleFactory: (function(!Element, !MDCRippleFoundation): !MDCRipple), foundation: !MDCRippleFoundation)`
```

Comment:
I think we tend to not include the `!` in our notation in the READMEs which follows TypeScript syntax.